### PR TITLE
Fix batch normalization

### PR DESCRIPTION
**Pull request by okuta**
This PR fix bug when use `volatile="on"` and ~~`test=True`~~ `test=False`.

### CODE REVIEW

File: tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py [hunk 1]
```diff
@@ -53,6 +55,19 @@ def test_forward_gpu(self):
         self.link.to_gpu()
         self.check_forward(cuda.to_gpu(self.x))
 
+    @condition.retry(3)
+    def test_forward_cpu_volatile(self):
+        self.volatile = 'on'
+        self.link.test = True
```
Comment (delta2323) [inline tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: Does `BatchNormalization` or `Link` has `test` attribute? Otherwise, this assignment does not have any effect.

File: tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py [hunk 2]
```diff
@@ -30,8 +30,10 @@ def setUp(self):
         self.x = numpy.random.uniform(-1, 1, (7, 3)).astype(numpy.float32)
         self.gy = numpy.random.uniform(-1, 1, (7, 3)).astype(numpy.float32)
 
+        self.volatile = 'off'
+
     def check_forward(self, x_data):
-        x = chainer.Variable(x_data)
+        x = chainer.Variable(x_data, volatile=self.volatile)
         y = self.link(x)
```
Comment (delta2323) [inline tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: I think we must pass `test` option to this __call__.

**Comment by delta2323** [comment]
How about make the test parameterized with respect to `volatile` and `test`?

**Comment by okuta** [comment]
@delta2323 I fixed the code.

**Comment by delta2323** [comment]
LGTM :+1:
